### ISSUE TITLE
[import/no-unresolved] not working for index.js

### ISSUE THREAD

**Issue by a-x-** [benmosher/eslint-plugin-import#1292]
similar but different: https://github.com/benmosher/eslint-plugin-import/issues/1124

<img width="1025" alt="screenshot 2019-02-19 at 20 45 03" src="https://user-images.githubusercontent.com/6201068/53036572-104a9d00-3489-11e9-9ff3-bf0f8708a323.png">

**file struct**:
```
utils/
    index.js
components/
    Foo.jsx
.eslintrc.js
webpack.config.js
```

**utils/index.js**:
```js
export function getPromise () { return 42; }
```

**components/Foo.jsx**:
```jsx
import { getPromise } from 'utils';
```

**.eslintrc**:
```js
{
  "parser": "babel-eslint",
  "plugins": [
    "import",
  ],
  "settings": {
    "import/resolver": {
      "webpack": {
        "config": path.resolve(__dirname, "webpack.config.js"),
      }
    },
  },
  "rules": {
    "import/no-unresolved": ["error", {commonjs: true, caseSensitive: true}],
    // ...
  }
}
```

**webpack.config.js**:
```js
{
  context: root,
  resolve: {
    modules: [root, 'node_modules'],
    extensions: ['.js', '.jsx', '.json', '.coffee', '.cjsx'],
  },
  // ...
}
```

**versions**: all are latest

**Comment by ljharb** [comment]
How does `utils` work when it's not inside node_modules?

**Comment by a-x-** [comment]
due to webpack aliasing and module paths:
`modules: [root, 'node_modules']`
`root` here contains utils dir along with all the shown fs structure.

**Comment by a-x-** [comment]
No one wanna write awful `'../../../../MyComponent'` imports, right? :)

**Comment by ljharb** [comment]
I do - I think they're a correct and proper indication that I have code that needs extracting into a separate package :-)

In this case, tho, the error message is warning you about capitalization. If you import from `utils/index.js`, what happens?

**Comment by a-x-** [comment]
utils/index.js and utils/index work
capitalization is correct as I can check (`ls src`)

**Comment by a-x-** [comment]
> extracting into a separate package
Yes, but package can be local (that is well-known practice called Vendoring as I know),
so I setup webpack look for modules not in node_modules only

dependency path variable is common computer engineering practice too

**Comment by ljharb** [comment]
You can and should vendor into node_modules itself and use `bundledDependencies` to tell npm to ignore that folder; it’s not a good practice imo to create alternate or different directories for third party code.

However, the `utils` issue is interesting. I’d still expect it to work and the implication is that there’s either a bug we need to fix, or that there’s something weird with capitalization of your files or case sensitivity of your machine.

**Comment by a-x-** [comment]
That's interesting solution... We'll think about it.

I have default macOS config. MacOS suggests case-insensitive FS setting by default.
Anyway, I can check case in my github repo. so utils/index.js is not capitalized 🤔

**Comment by a-x-** [comment]
> That's interesting solution... We'll think about it.

really we're think that both tricks with moving all our directories into node_modules,
and using paths with ../../../ are highly weird.

all our friends use relative paths as single useful way to fight the path complexity (simplify refactorings, decrease code blowing and etc.)

I can help fix that maybe, if you can help with details

**Comment by a-x-** [comment]
Currently I disabled caseSensitive option, but it's not ideal solution for us

**Comment by ljharb** [comment]
Using absolute paths tends to couple things to specific developer’s machines (since all your files should only be inside `$HOME`; it’s not a good practice.

There’s definitely a bug here, and yes it’d be great to get a PR for it.

**Comment by a-x-** [comment]
Excuse my inaccuracy

I'm not about `/Users/mxtnr/projects/project/blah/blah/Foo.jsx`
I'm about `blah/blah/Foo.jsx`
rather than `../../../blah/Foo.jsx`

**Comment by Germain67** [comment]
Any news on this? We encountered the same issue in our project.

**Comment by OoDeLally** [comment]
How do people solve this issue as of today?

**Comment by a-x-** [comment]
@OoDeLally, I fexed it for most cases with that workaround:
https://github.com/benmosher/eslint-plugin-import/issues/1292#issuecomment-469219706

but my project has not case mismatches.

so I wanna to get `caseSensitive: true` config back because some juniors can cause some naming/import case mismatching problems in the future.

**Comment by xianshenglu** [comment]
usually I'd like to write code like `import utils from '@/utils'` rather than `import utils from '@/utils/index.js'`.
